pkg/model/faucet/test/testenv.go

```diff
@@ -0,0 +1,353 @@
+		_ = te.StoreMessage(msg) // no need to release, since we remember all the messages for later cleanup
+		return nil
+	}
+
+	f := faucet.New(
+		defaultDaemon,
+		te.Storage(),
+		te.SyncManager(),
+		te.NetworkID(),
+		int(te.BelowMaxDepth()),
+		te.UTXOManager(),
+		faucetWallet.Address(),
+		faucetWallet.AddressSigner(),
+		tipselFunc,
+		te.PoWHandler,
+		storeMessageFunc,
+		faucet.WithHRPNetworkPrefix(iotago.PrefixTestnet),
+		faucet.WithAmount(faucetAmount),
+		faucet.WithSmallAmount(faucetSmallAmount),
+		faucet.WithMaxAddressBalance(faucetMaxAddressBalance),
+		faucet.WithMaxOutputCount(faucetMaxOutputCount),
+		faucet.WithIndexationMessage(faucetIndexationMessage),
+		faucet.WithBatchTimeout(faucetBatchTimeout),
+		faucet.WithPowWorkerCount(faucetPowWorkerCount),
+	)
+
+	faucetCtx, faucetCtxCancel := context.WithCancel(context.Background())
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		if err := f.RunFaucetLoop(faucetCtx, func() {
+			wg.Done()
+		}); err != nil && common.IsCriticalError(err) != nil {
+			require.NoError(t, err)
+		}
+	}()
+
+	// wait until faucet is initialized
+	wg.Wait()
+
+	// Connect the callbacks from the testsuite to the Faucet
+	te.ConfigureUTXOCallbacks(
+		nil,
+		nil,
+		func(confirmation *whiteflag.Confirmation) {
+			require.NoError(t, f.ApplyConfirmation(confirmation))
+		},
+		nil,
+	)
+
+	return &FaucetTestEnv{
+		t:               t,
+		TestEnv:         te,
+		GenesisWallet:   genesisWallet,
+		FaucetWallet:    faucetWallet,
+		Wallet1:         seed1Wallet,
+		Wallet2:         seed2Wallet,
+		Wallet3:         seed3Wallet,
+		Faucet:          f,
+		faucetCtxCancel: faucetCtxCancel,
+	}
+}
+
+func (env *FaucetTestEnv) ConfirmedMilestoneIndex() milestone.Index {
+	return env.TestEnv.SyncManager().ConfirmedMilestoneIndex()
+}
+
+func (env *FaucetTestEnv) Cleanup() {
+	if env.faucetCtxCancel != nil {
+		env.faucetCtxCancel()
+	}
+	env.TestEnv.CleanupTestEnvironment(true)
+}
+
+func (env *FaucetTestEnv) processFaucetRequests(preFlushFunc func() error) (hornet.MessageIDs, error) {
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	var tips hornet.MessageIDs
+	onFaucetIssuedMessage := events.NewClosure(func(messageID hornet.MessageID) {
+		tips = append(tips, messageID)
+		wg.Done()
+	})
+	env.Faucet.Events.IssuedMessage.Attach(onFaucetIssuedMessage)
+	defer env.Faucet.Events.IssuedMessage.Detach(onFaucetIssuedMessage)
+
+	if preFlushFunc != nil {
+		if err := preFlushFunc(); err != nil {
+			return nil, err
+		}
+	}
+
+	env.Faucet.FlushRequests()
+
+	chanDone := make(chan struct{})
+
+	go func() {
+		wg.Wait()
+		close(chanDone)
+	}()
+
+	select {
+	case <-chanDone:
+	case <-time.After(1 * time.Second):
+		env.t.Error("attachment of faucet message took too long")
+	}
+
+	return tips, nil
+}
+
+// RequestFunds sends requests to the faucet and waits until the next faucet message is issued.
+func (env *FaucetTestEnv) RequestFunds(wallets []*utils.HDWallet) (hornet.MessageIDs, error) {
```

Comment:
if you make wallets variadic, then you don't need to always wrap into an array when you want to call it with just 1